rpc-broker/src/policy.c

```diff
@@ -0,0 +1,258 @@
+/*
+ Copyright (c) 2018 AIS, Inc.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#include "rpc-broker.h"
+
+
+static int create_rule(struct rule *current, char *rule)
+{
+    char *token;
+    const char *delimiter;
+
+    if (!rule || rule[0] == '\0')
+        return -1;
+
+    current->rule_string = strdup(rule);
+
+    delimiter = " ";
+    token = strtok(rule, delimiter);
+
+    if (!token)
+        return -1;
+
+    current->policy = token[0] == 'a';
```

Comment:
This line seems to mean "Set the type of rule (allow/deny) depending on the first character of the rule line, if it's an 'a' it must be allow, since deny doesn't start with an 'a'". This is hard to deduct by looking at `current->policy = token[0] == 'a';`... I wouldn't mix tests and assignations in the same line, maybe add a comment, and consider making "policy" a boolean, or an enum, like in https://github.com/OpenXT/vusb-daemon/blob/master/src/policy.h